bin/testEnvLocalK8S.sh

```diff
@@ -76,14 +78,24 @@ function getDeps() {
      elif [ -f /tmp/apiserver/kube-apiserver ] ; then
        ln -s /tmp/apiserver/kube-apiserver $TOP/bin/
      else
-       curl -Lo ${TOP}/bin/kube-apiserver https://storage.googleapis.com/kubernetes-release/release/${K8S_VER}/bin/linux/amd64/kube-apiserver && chmod +x ${TOP}/bin/kube-apiserver
+       # bucket doesn't contain a kube-apiserver for darwin
+       curl -Lo ${TOP}/bin/kube-apiserver https://storage.googleapis.com/kubernetes-release/release/${K8S_VER}/bin/${LOCAL_GOOS}/amd64/kube-apiserver && chmod +x ${TOP}/bin/kube-apiserver
```

Comment:
eventually we need to nuke those multi curls across the code base (we already removed the ones in init.sh, replaced by vendor like they should have been from day 1) and have all the dependencies in 1 single vendor like download or image dependency

Reply:
I agree it's be nice (I'd tend towards also porting the file to make), though I'm tending towards smaller PRs that are more likely for me to get to completion now that I've got another day job (despite my other change that I'll probably turn into a PR once vendor has the newer fortio).  I did try to run a "make test" on my Linux box and first ran the make target that runs this script, but I still got failures (mostly from a non-present kubeconfig file, and creating an empty one just changes the complain message but the overall result is still a failure).

Reply:
a (somewhat low pri) test of our makefile infra is if can run from "docker run ubuntu:xenial" (+ git + go + docker - I have such an image actually it's `istio/fortio.build:v6` but we should make one specifically for istio "install from scratch is possible" tests )

Reply:
It'll probably also need fpm, at least if you care about the debian.  Costin has previously advocated for using the image that's used by CircleCI (the debian makefile has an optional target that tries to build the debian file from within a docker image).  There's yet another image used by prow/Cloud Builder.  So, everybody has their favorite. I've tried to stay out of the image debates and focused on getting the Makefile to check/resolve its own dependencies that can be conveniently run locally (basically excludes git, go, docker, and probably fpm too),  or at least have it check and complain when something's missing.  I just don't know yet what's actually needed to get "make test" to pass (either in theory or actuality).

Reply:
sgtm